engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,394 @@
+        this.tableDef = tableDef;
+
+        final String outputColumnName = pair.leftColumn;
+
+        formulaColumn = FormulaColumn.createFormulaColumn(outputColumnName, formula);
+
+        final Map<String, ColumnDefinition<?>> columnDefinitionMap = tableDef.getColumnNameMap();
+
+        // Create a column definition map composed of vector types for the formula.
+        // TODO: do this ahead of time and provide it as an alternate table definition?
+        final Map<String, ColumnDefinition<?>> vectorColumnNameMap = new HashMap<>();
+        columnDefinitionMap.entrySet().forEach(e -> {
+            final ColumnDefinition<?> columnDef = ColumnDefinition.fromGenericType(
+                    e.getKey(), VectorFactory.forElementType(e.getValue().getDataType()).vectorType());
+            vectorColumnNameMap.put(e.getKey(), columnDef);
+        });
+
+        // Get the input column names and data types from the formula.
+        inputColumnNames = formulaColumn.initDef(vectorColumnNameMap, compilationProcessor).toArray(String[]::new);
+        inputColumnTypes = new Class[inputColumnNames.length];
+        inputComponentTypes = new Class[inputColumnNames.length];
+        inputVectorTypes = new Class[inputColumnNames.length];
+
+        for (int i = 0; i < inputColumnNames.length; i++) {
+            final ColumnDefinition<?> columnDef = columnDefinitionMap.get(inputColumnNames[i]);
+            inputColumnTypes[i] = columnDef.getDataType();
+            inputComponentTypes[i] = columnDef.getComponentType();
+            inputVectorTypes[i] = vectorColumnNameMap.get(inputColumnNames[i]).getDataType();
+        }
+    }
+
+    @Override
+    public UpdateByOperator copy() {
+        return null;
+    }
+
+    @Override
+    public void initializeSources(@NotNull final Table source, @Nullable final RowRedirection rowRedirection) {
+        this.rowRedirection = rowRedirection;
+
+        if (rowRedirection != null) {
+            // region create-dense
+            maybeInnerSource = ArrayBackedColumnSource.getMemoryColumnSource(0, formulaColumn.getReturnedType());
+            // endregion create-dense
+            outputSource = WritableRedirectedColumnSource.maybeRedirect(rowRedirection, maybeInnerSource, 0);
+        } else {
+            maybeInnerSource = null;
+            // region create-sparse
+            outputSource = SparseArrayColumnSource.getSparseMemoryColumnSource(0, formulaColumn.getReturnedType());
+            // endregion create-sparse
+        }
+
+        primitiveOutputSource = ReinterpretUtils.maybeConvertToWritablePrimitive(outputSource);
+
+        outputChunkType = primitiveOutputSource.getChunkType();
+    }
+
+    protected static IntConsumer getChunkSetter(
+            final WritableChunk<? extends Values> valueChunk,
+            final ColumnSource<?> formulaOutputSource) {
+        final ChunkType chunkType = valueChunk.getChunkType();
+        if (chunkType == ChunkType.Boolean) {
```

Comment:
Looks like a lot of duplicated code from `BaseRollingFormulaOperator::getChunkSetter`. Maybe there's an opportunity to reuse?

Reply:
We intend to deprecate the single column RollingFormula in future releases so intermingling the classes might be wasted work and/or create future work.

Reply:
Makes sense